spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -421,6 +421,23 @@ def count
       end.to raise_error(/#let or #subject called without a block/)
     end
 
+    def silence_warnings
+      old_verbose, $VERBOSE = $VERBOSE, nil
+      yield
+    ensure
+      $VERBOSE = old_verbose
+    end
+
+    it 'emits a warning when declared multiple times' do
+      silence_warnings do
+        expect(RSpec).to receive(:warning).with(/defined multiple times through let in the same context/)
+        RSpec.describe do
+          let(:foo) { 'first value' }
+          let(:foo) { 'second value' }
+        end
+      end
+    end
```

Comment:
Instead of `silence_warnings { expect(RSpec).to receive(:warning); ... }`, WDYT of this?

``` ruby
it 'emits a warning when declared multiple times' do
  line = nil
  expect {
    RSpec.describe do
      let(:foo) { 'first value' }
      let(:foo) { 'second value' }; line = __LINE__
    end
  }.to output(a_string_including("#foo", "let(:foo)", "definition at #{__FILE__}:#{line}")).to_stderr
end
```

I think it's better because it doesn't require `silence_warnings` and it's less coupled to the implementation -- it simply passes if the snippet is written to stderr, regardless of how it got there (e.g. via `Kernel#warn` or `RSpec.warning` or whatever).  `a_string_including` lets you easily match multiple snippets, including "definition at #{line}" which is important so that the user is notified of the line where the problem is.